### ISSUE TITLE
Send email confirmation now breaks when request=None

### ISSUE THREAD

**Issue by rptmat57** [pennersr/django-allauth#3245]
This issue was introduced in version 0.52.0, I used to be able to call `send_email_confirmation(None, user)` before (I use it on a user post save and there is no request object available)
The issue comes down to a check in `ratelimit`:
```python
if request.method == "GET" or not amount or not duration:
    pass
```

Which I would argue should be changed to:
```python
if not request or request.method == "GET" or not amount or not duration:
    pass
```

**Comment by iarp** [comment]
Until something else changes, you can supply your own `ACCOUNT_ADAPTER` and override `should_send_confirmation_mail` as that is where the request is being used and throwing its error.

https://github.com/pennersr/django-allauth/blob/master/allauth/account/adapter.py#L517

**Comment by rptmat57** [comment]
I thought about that, but the problem is that the check is done inside the ratelimit `consume` function.
So I would have to override `should_send_confirmation_email` just to provide my own `consume` function, which in turn would be exactly the same as the original except for that one simple check.
Feels a bit overkill.

In the meantime, I decided to simply disable `EMAIL_CONFIRMATION_HMAC`, but I don't feel like it's a great solution either.

**Comment by pennersr** [comment]
Closed via 354f0cd6

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3245
